hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenRejectHandler.java

```diff
@@ -62,7 +70,51 @@ public void handle(@NonNull final HandleContext context) throws HandleException
     @NonNull
     @Override
     public Fees calculateFees(final FeeContext feeContext) {
-        // Todo: Implement the logic for calculating fees
-        return Fees.FREE;
+        final var body = feeContext.body();
```

Comment:
Why do we need all this logic, if we are diispatching a `CryptoTransfer` internally ? Wouldn't a `CryptoTransfer.calculateFees` be sufficient ?

Reply:
Hey Neeha!  
If I am getting you right about what do you refer to as "dispatching CryptoTransfer internally"
I am not sure if that is possible without breaking the flow. Isn't every handler's functionality currently confined to itself, i.e., no HAPI operation dispatches another one?

Also we have to skip some checks that are specific to the TokenReject functionality, like skipping requiredReceiverSig or transferring even if the token is paused.

Reply:
Hey @MrValioBg , I think if we call `dispatchSyntheticTransaction` it computes fees also. We can extract common functionality imo.

Reply:
If we use  the `dispatchSyntheticTransaction` and compute fees from it, what we will do about the Fees of the TokenReject, if it fails too early before dispatching?
[Docs](https://github.com/hashgraph/hedera-services/pull/12786/files#diff-345e91521530c2849705b8d557dc66a644189322a8cc02910e1b5ea11cd73929R102) mentions that the fees should be the same as of a basic transfer without custom fees.

On another note, if we call `dispatchSyntheticTransaction`, the checks for required signatures for the receiver and the paused token, I believe, are in the handler itself, so handling it specifically for TokenReject could be challenging.

As you mentioned we could definitely extract common logic from the transfer handler logic and use it, but I am not quite sure about dispatching it, because of the cases I mentioned.

Reply:
I see. We should atleast extract common logic from `CryptoTransferHandler` instead of duplicating here.

Reply:
Definitely.